### PR TITLE
updated cd info on _commands/basic page

### LINKED ISSUE #234: Update description of "cd" to clarify moving out of directory (..)

### PR DESCRIPTION
**Pull request by AlborzB**
# Advanced HW 7 submission

Closes #234

Name: Alborz Bral
Uniqname: abral

### CODE REVIEW

File: _commands/basics/cd.md [hunk 1]
```diff
@@ -16,7 +16,87 @@ $ cd /home
 
 ### Useful Options / Examples
 
+
 #### `cd -`
```
Comment (AlborzB) [inline _commands/basics/cd.md]: Thanks for the feedback, I updated the page to accommodate this and pushed my changes to the forked repo. I am not sure if I need to submit another pull request so let me know if you cannot see the changes.

Comment (mterwill) [inline _commands/basics/cd.md]: Looks good, thanks!

File: _commands/basics/cd.md [hunk 2]
```diff
@@ -16,7 +16,88 @@ $ cd /home
...
+
+Of course, this accomplishes nothing; however, there are other uses for the `.` character pertaining to the `cd` command:
 
+`.` can mean "source", as in the following example
+
+~~~ bash
+./my_script
+~~~ 
+
+However, note that when we use `.`, it runs "my_script" as an executable, and launches a new shell to do so. On the other hand, if we were to use `source` instead, like so:
```
Comment (mterwill) [inline _commands/basics/cd.md]: Do you think this introduces more confusion than it helps? The dot can mean lots of things, and if you're going to include this it'd be good to disambiguate a little better:

__As a POSIX shell construct__
Synonymous with `source`, as you mentioned. Used to execute commands in the current shell.

__As a filesystem construct__
`.` and `..` are special filesystem constructs that are [hard links](https://en.wikipedia.org/wiki/Hard_link) to the current and parent directories, respectively.

__As a "dotfile"__
There's a convention that filenames beginning with `.` are hidden by default. (e.g. `.git`)

---

__So__, something like this is perfectly valid:

```
$ . ./.my_script.sh
```

1. Our filename is called `.my_script.sh`.
2. The reason we have to run `./.my_script.sh` is because `.`, or the current working directory, is not in `PATH`. This is _exactly_ the same as the full path, e.g. `/Users/mterwilliger/Desktop/.my_script.sh`, and has nothing to do with the shell. The reason our shell does not search our current directory for executable is security related: we don't want to automatically execute arbitrary programs that happen to be in the current (potentially untrusted) directory.
3. We then want to execute this inline, instead of in a subshell.

Comment (mterwill) [inline _commands/basics/cd.md]: Even

```
$ . ../...
```

`source` the file named `...` in the parent (`..`) directory

Comment (mterwill) [inline _commands/basics/cd.md]: Makes sense. How about if I pull my comment into a separate reference page and you can link to that?

Comment (AlborzB) [inline _commands/basics/cd.md]: Sure, that works! I'll just include a link to that page then, and a brief description of what is on it, if that sounds okay.

Comment (mterwill) [inline _commands/basics/cd.md]: Great! Just pushed. Live at https://c4cs.github.io/commands/dot.

File: _commands/basics/cd.md (hunk 1 above)
Comment (AlborzB) [inline _commands/basics/cd.md]: I just pushed the updated file. One thing I wasn't able to figure out is how to actually create a hyperlink out of a url in the text file, so the link doesn't show up blue and can't be followed with a click. I know this is a small detail but I can go ahead and modify that too, since it would make the page look nicer, given that it is possible to do so.

File: _commands/basics/cd.md [hunk 3]
```diff
@@ -60,22 +60,7 @@ Alborz-Brals-MacBook-Pro:basics abral$ pwd
...
-
-However, note that when we use `.`, it runs "my_script" as an executable, and launches a new shell to do so. On the other hand, if we were to use `source` instead, like so:
-
-~~~ bash
-source my_script
-~~~
-
-this command would read and execute the commands within the current shell environment
-
+Of course, this accomplishes nothing; however, there are other uses for the `.` character pertaining to the `cd` command. Some of the more common uses can be found on the dot(.) page of the C4CS website: https://c4cs.github.io/commands/dot
```
Comment (mterwill) [inline _commands/basics/cd.md]: In markdown you have to manually specify links in the form:

```
[C4CS website](https://c4cs.github.io/commands/dot)
```

Comment (AlborzB) [inline _commands/basics/cd.md]: Got it, fixed that and pushed the changes.

**Comment by mterwill** [comment]
Thanks for fixing that up! Will merge + give you credit after tests pass.

**Comment by AlborzB** [comment]
thanks!
